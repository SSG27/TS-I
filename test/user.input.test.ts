import React from 'react';
import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';
import App from '../src/App';

it('updates state on user input', () => {
  // Create a snapshot for the initial rendering
  const component = renderer.create(<App />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // Manually trigger user input for movie title
  renderer.act(() => {
    userEvent.type(tree.children[1].children[5], 'dunkirk');
  });

  // Re-rendering after user input
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // Manually trigger user input for country code
  renderer.act(() => {
    userEvent.type(tree.children[1].children[5], 'gb');
    userEvent.click(tree.children[5]);
  });

  // Re-rendering after user input
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
