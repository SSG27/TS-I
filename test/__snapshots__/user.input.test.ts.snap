// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`updates state on user input 1`] = `
<section
  style={
    {
      "backgroundColor": "#1F2739",
      "color": "#fff",
      "fontFamily": "-apple-system",
      "fontSize": "1rem",
      "fontWeight": 1.5,
      "lineHeight": 1.5,
      "padding": "0 2em",
    }
  }
>
  <div
    style={
      {
        "border": "1px solid #e6e6e6",
        "margin": "0 auto",
        "marginTop": "50px",
        "maxWidth": "950px",
        "padding": "45px 25px 50px 25px",
        "textAlign": "center",
      }
    }
  >
    <img
      alt=""
      src={
        {
          "process": [Function],
        }
      }
      style={
        {
          "margin": "-90px auto 30px",
          "marginBottom": "0",
          "objectFit": "cover",
          "width": "100px",
        }
      }
    />
    <div>
      <p
        style={
          {
            "color": "#FFF842",
            "fontSize": "1.375rem",
            "fontWeight": 300,
            "lineHeight": 1.5,
            "marginBottom": "25px",
          }
        }
      >
        Your one stop search for movies to watch.
      </p>
    </div>
    <p
      style={
        {
          "color": "#FB667A",
          "fontSize": "1rem",
          "fontWeight": 600,
          "marginBottom": "0",
        }
      }
    >
      Made By
      <span
        style={
          {
            "fontWeight": 400,
          }
        }
      >
         · Sanju Ghandhi
      </span>
    </p>
  </div>
  <form
    onSubmit={[Function]}
  >
    <br />
    <label>
      Enter the movie title:
      <input
        name="movietitle"
        onChange={[Function]}
        type="text"
        value=""
      />
    </label>
    <br />
    <label>
      Enter the two letter country code (e.g gb):
      <input
        name="moviecountry"
        onChange={[Function]}
        type="text"
        value=""
      />
    </label>
    <br />
    <input
      type="submit"
    />
  </form>
  <button
    onClick={[Function]}
  >
    Generate Text Content
  </button>
  <h2>
    Generated Text Content:
  </h2>
  <pre />
  <button
    onClick={[Function]}
  >
    Save Text Content
  </button>
</section>
`;

exports[`updates state on user input 2`] = `
<section
  style={
    {
      "backgroundColor": "#1F2739",
      "color": "#fff",
      "fontFamily": "-apple-system",
      "fontSize": "1rem",
      "fontWeight": 1.5,
      "lineHeight": 1.5,
      "padding": "0 2em",
    }
  }
>
  <div
    style={
      {
        "border": "1px solid #e6e6e6",
        "margin": "0 auto",
        "marginTop": "50px",
        "maxWidth": "950px",
        "padding": "45px 25px 50px 25px",
        "textAlign": "center",
      }
    }
  >
    <img
      alt=""
      src={
        {
          "process": [Function],
        }
      }
      style={
        {
          "margin": "-90px auto 30px",
          "marginBottom": "0",
          "objectFit": "cover",
          "width": "100px",
        }
      }
    />
    <div>
      <p
        style={
          {
            "color": "#FFF842",
            "fontSize": "1.375rem",
            "fontWeight": 300,
            "lineHeight": 1.5,
            "marginBottom": "25px",
          }
        }
      >
        Your one stop search for movies to watch.
      </p>
    </div>
    <p
      style={
        {
          "color": "#FB667A",
          "fontSize": "1rem",
          "fontWeight": 600,
          "marginBottom": "0",
        }
      }
    >
      Made By
      <span
        style={
          {
            "fontWeight": 400,
          }
        }
      >
         · Sanju Ghandhi
      </span>
    </p>
  </div>
  <form
    onSubmit={[Function]}
  >
    <br />
    <label>
      Enter the movie title:
      <input
        name="movietitle"
        onChange={[Function]}
        type="text"
        value=""
      />
    </label>
    <br />
    <label>
      Enter the two letter country code (e.g gb):
      <input
        name="moviecountry"
        onChange={[Function]}
        type="text"
        value=""
      />
    </label>
    <br />
    <input
      type="submit"
    />
  </form>
  <button
    onClick={[Function]}
  >
    Generate Text Content
  </button>
  <h2>
    Generated Text Content:
  </h2>
  <pre />
  <button
    onClick={[Function]}
  >
    Save Text Content
  </button>
</section>
`;

exports[`updates state on user input 3`] = `
<section
  style={
    {
      "backgroundColor": "#1F2739",
      "color": "#fff",
      "fontFamily": "-apple-system",
      "fontSize": "1rem",
      "fontWeight": 1.5,
      "lineHeight": 1.5,
      "padding": "0 2em",
    }
  }
>
  <div
    style={
      {
        "border": "1px solid #e6e6e6",
        "margin": "0 auto",
        "marginTop": "50px",
        "maxWidth": "950px",
        "padding": "45px 25px 50px 25px",
        "textAlign": "center",
      }
    }
  >
    <img
      alt=""
      src={
        {
          "process": [Function],
        }
      }
      style={
        {
          "margin": "-90px auto 30px",
          "marginBottom": "0",
          "objectFit": "cover",
          "width": "100px",
        }
      }
    />
    <div>
      <p
        style={
          {
            "color": "#FFF842",
            "fontSize": "1.375rem",
            "fontWeight": 300,
            "lineHeight": 1.5,
            "marginBottom": "25px",
          }
        }
      >
        Your one stop search for movies to watch.
      </p>
    </div>
    <p
      style={
        {
          "color": "#FB667A",
          "fontSize": "1rem",
          "fontWeight": 600,
          "marginBottom": "0",
        }
      }
    >
      Made By
      <span
        style={
          {
            "fontWeight": 400,
          }
        }
      >
         · Sanju Ghandhi
      </span>
    </p>
  </div>
  <form
    onSubmit={[Function]}
  >
    <br />
    <label>
      Enter the movie title:
      <input
        name="movietitle"
        onChange={[Function]}
        type="text"
        value=""
      />
    </label>
    <br />
    <label>
      Enter the two letter country code (e.g gb):
      <input
        name="moviecountry"
        onChange={[Function]}
        type="text"
        value=""
      />
    </label>
    <br />
    <input
      type="submit"
    />
  </form>
  <button
    onClick={[Function]}
  >
    Generate Text Content
  </button>
  <h2>
    Generated Text Content:
  </h2>
  <pre />
  <button
    onClick={[Function]}
  >
    Save Text Content
  </button>
</section>
`;
